TypeScript errors from assigning a function with params to VOID

The following TypeScript errors are caused by `ol/source/Source~forEachFeatureAtCoordinate` being assigned to `ol/functions~VOID`. I would expect `tsc` to handle this better, but it seems like something that could also be fixed in OpenLayers.

```
src/ol/renderer/canvas/IntermediateCanvas.js:101:12 - error TS2554: Expected 0 arguments, but got 6.

101     return source.forEachFeatureAtCoordinate(
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
102       coordinate, resolution, rotation, hitTolerance, skippedFeatureUids,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
108         return callback.call(thisArg, feature, layer);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
109       });
    ~~~~~~~~

src/ol/renderer/webgl/ImageLayer.js:87:12 - error TS2554: Expected 0 arguments, but got 6.

 87     return source.forEachFeatureAtCoordinate(
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 88       coordinate, resolution, rotation, hitTolerance, skippedFeatureUids,
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 95         return callback.call(thisArg, feature, layer);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 96       });
    ~~~~~~~~
```

It seems the default value of `ol/source/Source~forEachFeatureAtCoordinate` could be changed to an empty function (instead of `VOID`), but that would break the following related conditions:
https://github.com/openlayers/openlayers/blob/43ca7642938c9e8756e789b68d3aa2256ff51cd4/src/ol/renderer/canvas/IntermediateCanvas.js#L120-L123

https://github.com/openlayers/openlayers/blob/43ca7642938c9e8756e789b68d3aa2256ff51cd4/src/ol/renderer/webgl/ImageLayer.js#L233-L243

Is there a recommended approach to refactoring those?

--- a/src/ol/renderer/canvas/IntermediateCanvas.js
+++ b/src/ol/renderer/canvas/IntermediateCanvas.js
@@ -4,7 +4,6 @@
 import {scale as scaleCoordinate} from '../../coordinate.js';
 import {createCanvasContext2D} from '../../dom.js';
 import {containsExtent, intersects} from '../../extent.js';
-import {VOID} from '../../functions.js';
 import CanvasLayerRenderer from '../canvas/Layer.js';
 import {create as createTransform, apply as applyTransform} from '../../transform.js';
 
@@ -92,52 +91,26 @@ class IntermediateCanvasRenderer extends CanvasLayerRenderer {
   /**
    * @inheritDoc
    */
-  forEachFeatureAtCoordinate(coordinate, frameState, hitTolerance, callback, thisArg) {
-    const layer = this.getLayer();
-    const source = layer.getSource();
-    const resolution = frameState.viewState.resolution;
-    const rotation = frameState.viewState.rotation;
-    const skippedFeatureUids = frameState.skippedFeatureUids;
-    return source.forEachFeatureAtCoordinate(
-      coordinate, resolution, rotation, hitTolerance, skippedFeatureUids,
-      /**
-       * @param {import("../../Feature.js").FeatureLike} feature Feature.
-       * @return {?} Callback result.
-       */
-      function(feature) {
-        return callback.call(thisArg, feature, layer);
-      });
-  }
-
-  /**
-   * @inheritDoc
-   */
   forEachLayerAtCoordinate(coordinate, frameState, hitTolerance, callback, thisArg) {
     if (!this.getImage()) {
       return undefined;
     }
 
-    if (this.getLayer().getSource().forEachFeatureAtCoordinate !== VOID) {
-      // for ImageCanvas sources use the original hit-detection logic,
-      // so that for example also transparent polygons are detected
-      return super.forEachLayerAtCoordinate(coordinate, frameState, hitTolerance, callback, thisArg);
+    const pixel = applyTransform(this.coordinateToCanvasPixelTransform, coordinate.slice());
+    scaleCoordinate(pixel, frameState.viewState.resolution / this.renderedResolution);
+
+    if (!this.hitCanvasContext_) {
+      this.hitCanvasContext_ = createCanvasContext2D(1, 1);
+    }
+
+    this.hitCanvasContext_.clearRect(0, 0, 1, 1);
+    this.hitCanvasContext_.drawImage(this.getImage(), pixel[0], pixel[1], 1, 1, 0, 0, 1, 1);
+
+    const imageData = this.hitCanvasContext_.getImageData(0, 0, 1, 1).data;
+    if (imageData[3] > 0) {
+      return callback.call(thisArg, this.getLayer(), imageData);
     } else {
-      const pixel = applyTransform(this.coordinateToCanvasPixelTransform, coordinate.slice());
-      scaleCoordinate(pixel, frameState.viewState.resolution / this.renderedResolution);
-
-      if (!this.hitCanvasContext_) {
-        this.hitCanvasContext_ = createCanvasContext2D(1, 1);
-      }
-
-      this.hitCanvasContext_.clearRect(0, 0, 1, 1);
-      this.hitCanvasContext_.drawImage(this.getImage(), pixel[0], pixel[1], 1, 1, 0, 0, 1, 1);
-
-      const imageData = this.hitCanvasContext_.getImageData(0, 0, 1, 1).data;
-      if (imageData[3] > 0) {
-        return callback.call(thisArg, this.getLayer(), imageData);
-      } else {
-        return undefined;
-      }
+      return undefined;
     }
   }
 }

--- a/src/ol/renderer/webgl/ImageLayer.js
+++ b/src/ol/renderer/webgl/ImageLayer.js
@@ -2,7 +2,7 @@
  * @module ol/renderer/webgl/ImageLayer
  */
 import {ENABLE_RASTER_REPROJECTION} from '../../reproj/common.js';
-import {TRUE, VOID} from '../../functions.js';
+import {TRUE} from '../../functions.js';
 import LayerType from '../../LayerType.js';
 import ViewHint from '../../ViewHint.js';
 import {createCanvasContext2D} from '../../dom.js';
@@ -230,50 +230,36 @@ class WebGLImageLayerRenderer extends WebGLLayerRenderer {
       return undefined;
     }
 
-    if (this.getLayer().getSource().forEachFeatureAtCoordinate !== VOID) {
-      // for ImageCanvas sources use the original hit-detection logic,
-      // so that for example also transparent polygons are detected
-      const coordinate = applyTransform(
-        frameState.pixelToCoordinateTransform, pixel.slice());
-      const hasFeature = this.forEachFeatureAtCoordinate(coordinate, frameState, 0, TRUE, this);
+    const imageSize =
+        [this.image_.getImage().width, this.image_.getImage().height];
 
-      if (hasFeature) {
-        return callback.call(thisArg, this.getLayer(), null);
-      } else {
-        return undefined;
-      }
+    if (!this.hitTransformationMatrix_) {
+      this.hitTransformationMatrix_ = this.getHitTransformationMatrix_(
+        frameState.size, imageSize);
+    }
+
+    const pixelOnFrameBuffer = applyTransform(
+      this.hitTransformationMatrix_, pixel.slice());
+
+    if (pixelOnFrameBuffer[0] < 0 || pixelOnFrameBuffer[0] > imageSize[0] ||
+        pixelOnFrameBuffer[1] < 0 || pixelOnFrameBuffer[1] > imageSize[1]) {
+      // outside the image, no need to check
+      return undefined;
+    }
+
+    if (!this.hitCanvasContext_) {
+      this.hitCanvasContext_ = createCanvasContext2D(1, 1);
+    }
+
+    this.hitCanvasContext_.clearRect(0, 0, 1, 1);
+    this.hitCanvasContext_.drawImage(this.image_.getImage(),
+      pixelOnFrameBuffer[0], pixelOnFrameBuffer[1], 1, 1, 0, 0, 1, 1);
+
+    const imageData = this.hitCanvasContext_.getImageData(0, 0, 1, 1).data;
+    if (imageData[3] > 0) {
+      return callback.call(thisArg, this.getLayer(), imageData);
     } else {
-      const imageSize =
-          [this.image_.getImage().width, this.image_.getImage().height];
-
-      if (!this.hitTransformationMatrix_) {
-        this.hitTransformationMatrix_ = this.getHitTransformationMatrix_(
-          frameState.size, imageSize);
-      }
-
-      const pixelOnFrameBuffer = applyTransform(
-        this.hitTransformationMatrix_, pixel.slice());
-
-      if (pixelOnFrameBuffer[0] < 0 || pixelOnFrameBuffer[0] > imageSize[0] ||
-          pixelOnFrameBuffer[1] < 0 || pixelOnFrameBuffer[1] > imageSize[1]) {
-        // outside the image, no need to check
-        return undefined;
-      }
-
-      if (!this.hitCanvasContext_) {
-        this.hitCanvasContext_ = createCanvasContext2D(1, 1);
-      }
-
-      this.hitCanvasContext_.clearRect(0, 0, 1, 1);
-      this.hitCanvasContext_.drawImage(this.image_.getImage(),
-        pixelOnFrameBuffer[0], pixelOnFrameBuffer[1], 1, 1, 0, 0, 1, 1);
-
-      const imageData = this.hitCanvasContext_.getImageData(0, 0, 1, 1).data;
-      if (imageData[3] > 0) {
-        return callback.call(thisArg, this.getLayer(), imageData);
-      } else {
-        return undefined;
-      }
+      return undefined;
     }
   }
 

--- a/src/ol/source/Source.js
+++ b/src/ol/source/Source.js
@@ -2,7 +2,6 @@
  * @module ol/source/Source
  */
 
-import {VOID} from '../functions.js';
 import BaseObject from '../Object.js';
 import {get as getProjection} from '../proj.js';
 import SourceState from './State.js';
@@ -172,18 +171,6 @@ class Source extends BaseObject {
     this.changed();
   }
 }
-
-/**
- * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
- * @param {number} resolution Resolution.
- * @param {number} rotation Rotation.
- * @param {number} hitTolerance Hit tolerance in pixels.
- * @param {Object<string, boolean>} skippedFeatureUids Skipped feature uids.
- * @param {function(import("../Feature.js").FeatureLike): T} callback Feature callback.
- * @return {T|void} Callback result.
- * @template T
- */
-Source.prototype.forEachFeatureAtCoordinate = VOID;
 
 
 /**

